IoTuring/Entity/Deployments/Wifi/Wifi.py

```diff
@@ -0,0 +1,174 @@
+        print("UpdateMac")
+
+    def GetWirelessStrength_Linux(self):
+        p = self.RunCommand(["iwconfig", self.configuredNic])
+        if p.stderr:
+            raise Exception("error during iwconfig")
+        elif p.stdout:
+            signalStrength = re.findall(f"Signal level=(-[0-9]+) dBm", p.stdout)
+        return signalStrength[0]
+
+    def GetWirelessStrength_Windows(self):
+        p = subprocess.Popen(
+            "netsh wlan show interfaces", stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        out = p.stdout.read().decode(errors="ignore")
+        if "Segnale" in out:  # Italian support
+            m = re.findall(
+                "Nome.*?:.*? ([A-z0-9 \-]*).*?Segnale.*?:.*?([0-9]*)%", out, re.DOTALL
+            )
+        elif "Signal" in out:  # English support
+            m = re.findall(
+                "Nome.*?:.*?([A-z0-9 ]*).*?Signal.*?:.*?([0-9]*)%", out, re.DOTALL
+            )
+        else:
+            self.Log(
+                self.Logger.LOG_ERROR,
+                "Can't get signal strength data in your region. Please open a Git issue with and show the output of this command in the CMD: 'netsh wlan show interfaces' ",
+            )
+            self.supports_signal_strength = False
+            # TODO Remove the signal strength topic for all the inets here !
+        p.communicate()
+        return m
+    def GetWirelessStrength_Mac(self):
+        raise NotImplementedError
+    
+    def Update(self):
+        self.specificUpdate()
+
+    @classmethod
+    def ConfigurationPreset(cls) -> MenuPreset:
+        # Get the choices for menu:
+        NIC_CHOICES = []
+
+        nics = psutil.net_if_addrs()
+
+        for nic in nics:
+            nicinfo = nics[nic]
+            # try to get nic ip
+            nicip4 = ""
+            nicip6 = ""
+            nicmac = ""
+            p = cls.RunCommand(["iwconfig", nic]) # type hint return missing in subprocess.CompletedProcess
+            if not bool(p.stdout): # if there is nothing in stdout for this nic it is no wireless interface
+                continue
+            for family in nicinfo:
+                if family.family==AddressFamily.AF_INET:
+                    nicip4 = family.address
+                elif family.family==AddressFamily.AF_INET6:
+                    nicip6 = family.address
+                elif family.family==psutil.AF_LINK:
+                    nicmac = family.address
+
+            NIC_CHOICES.append(
+                {
+                    "name": WIFI_CHOICE_STRING.format(
+                        nic, nicip4 if nicip4 else nicip6, nicmac # defaults to showing ipv4
+                    ),
+                    "value": nic,
+                }
+            )
+
+        preset = MenuPreset()
+        preset.AddEntry(
+            name="Interface to check",
+            key=CONFIG_KEY_WIFI,
+            mandatory=True,
+            question_type="select",
+            choices=NIC_CHOICES,
+        )
```

Comment:
if my logic does not find a wireless interface, there will be no choices passed in AddEntry, so it raises an exception. How can i errorhandle this cleanly? How can i communicate that to the user with the new interface?

Reply:
Add the logic for checking interfaces in `CheckSystemSupport`. Raise an exception if no interface found. `CheckSystemSupport` runs before the Config menu is displayed, if no nic found it won't show up there. It also runs before starting the loop.

Check how I changed it in Fanspeed: https://github.com/richibrics/IoTuring/blob/main/IoTuring/Entity/Deployments/Fanspeed/Fanspeed.py#L93

If you still need `RunCommand` there, it can be moved to OsD, and only wrapped around for logging in `Entity.RunCommand`.

Reply:
Okay, that makes sense and is a easy to grasp solution.

checking for iwconfig is necessary, although checking if the cli tool exist stil gives no information about any wifi nic being present. 
I'll find a way without running cli commands, for linux that may be possible. We'll see how windows and mac will handle that.

Reply:
has been resolved with moving RunCommand